File: HLC01_03Java/src/main/java/org/example/gestionBiblioteca01/Book.java
package org.example.gestionBiblioteca01;

public class Book {
    private String name;
    private String author;
    private int bookId;
    private boolean borrowed;


    public Book(String name, String author, int bookId) {
        this.name = name;
        this.author = author;
        this.bookId = bookId;
        this.borrowed = false;
    }


    public int getBookId() {
        return bookId;
    }


    public boolean isBorrowed() {
        return borrowed;
    }


    public void borrow() {
        this.borrowed = true;
    }


    public void unborrow() {
        this.borrowed = false;
    }


    @Override
    public String toString() {
        String status = borrowed ? "No Disponible" : "Disponible";
        return "\nNombre: " + name + "\nAutor: " + author + "\n" + status;
    }
}

File: HLC01_03Kotlin/src/main/kotlin/gestionBiblioteca01/Book.kt
package org.example.gestionBiblioteca01

class Book(
    private val name: String,
    private val author: String?,
    val bookId: Int,
) {

    private var borrowed = false


    fun isBorrowed(): Boolean {
        return borrowed
    }


    fun borrow() {
        borrowed = true
    }


    fun unborrow() {
        borrowed = false
    }


    override fun toString(): String {
        val unavailable = borrowed

        return if (unavailable) {
            "\nNombre: $name\nAutor: $author\nNo Disponible"
        } else {
            "\nNombre: $name\nAutor: $author\nDisponible"
        }

    }

}